client/post-editor/post-editor.jsx

```diff
@@ -574,10 +578,16 @@ export const PostEditor = React.createClass( {
 		this.setState( { isSaving: true } );
 	},
 
-	onPreview: function( event ) {
+	onPreview: function( action, event ) {
 		var status = 'draft',
 			previewPost;
 
+		if ( this.state.previewAction !== action ) {
+			this.setState( {
+				previewAction: action
+			} );
+		}
```

Comment:
Quick note: This comparison is helpful if the user does the following:

1. Cmd + Click on Preview to open a new tab.
2. Come back to the editor and click the Preview link like normal while the other tab is still open.

In that scenario, `this.state.previewAction` would still be set to `preview` so we're avoiding an extra `setState` call.